### PR TITLE
Add vnet-based provisioning options to cookbook

### PR DESCRIPTION
**Pull request by jfomhover**
## Purpose

Adds vnet based bicep provisioning to the cookbook.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: docs/provisioning/silo_vnet_newstorage.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Create a silo behind a vnet with a new private storage
+
+This tutorial applies in the case you want to create a **completely new storage** account in a federated learning silo in a given region.
```
Comment (thomasp-ms) [inline docs/provisioning/silo_vnet_newstorage.md:3]: maybe add a sentence or 2 explaining what kind of threats this protects from?

File: docs/provisioning/silo_vnet_newstorage.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+## Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Important: understand the design](#important-understand-the-design)
+- [Create a compute and storage pair for the silo](#need-support)
+  - [Option 1: one click deployment](#option-1-one-click-deployment)
+  - [Option 2: using az cli](#option-2-using-az-cli)
+- [Set permissions for the silo's compute to R/W from/to the orchestrator](#set-permissions-for-the-silos-compute-to-rw-fromto-the-orchestrator)
+
+## Prerequisites
```
Comment (thomasp-ms) [inline docs/provisioning/silo_vnet_newstorage.md:14]: In the Prerequisites, would it be worth repeating what level of permissions is required?

**Review by thomasp-ms — APPROVED** [review]

File: docs/provisioning/silo_vnet_newstorage.md (hunk 1 above)
Comment (jfomhover) [reply docs/provisioning/silo_vnet_newstorage.md:3]: Let me file an issue for that and implement later, I want to gather more info about what it actually protects from and make sure I get "authorivative" quotes from our docs.

File: docs/provisioning/silo_vnet_newstorage.md (hunk 2 above)
Comment (jfomhover) [reply docs/provisioning/silo_vnet_newstorage.md:14]: very good point, I applied here and also in the "open" scenario.

File: docs/provisioning/silo_vnet_newstorage.md (hunk 1 above)
Comment (jfomhover) [reply docs/provisioning/silo_vnet_newstorage.md:3]: https://github.com/Azure-Samples/azure-ml-federated-learning/issues/133

**Pull request merged**